crates/maybenot/src/time.rs

```diff
@@ -0,0 +1,61 @@
+use std::ops::AddAssign;
+
+/// Trait representing instants in time. Allows using maybenot frameworks with
+/// custom time sources. If you want to use maybenot with a different time source
+/// than `std::time::Instant`, implement this trait for your instant type, and the
+/// [`Duration`] trait for your corresponding duration type.
+pub trait Instant: Clone + Copy {
```

Comment:
Observation, non-blocking:

I'm not 100% sure that the "Copy" requirement on `Instant` and `Duration` should be strictly necessary. On the other hand,  I _do_ think that every sensible clock implementation would provide them, so maybe it's okay.

Reply:
It is necessary in the sense that if I remove it, then I must explicitly clone a bunch of instants across the codebase. So yes, I think it makes sense to require `Copy`. Any sane implementation of `Instant` will likely be a tiny trivially copyable type anyway.